### PR TITLE
Adds Ability To Pass In More OneSignal Options For Init

### PR DESCRIPTION
**Pull request by David-Melo**
Hello, Thanks for this package. I've been looking for something like this and I'm glad I've found it. 

I needed to pass in some additional options to the OneSignal Initializer to get it working with my setup.

I refactored the code so that instead of just passing in **appId** and **showNotifyButton** you could now pass in an **Options** object based on the [OneSignal Init Parameters](https://documentation.onesignal.com/docs/web-push-sdk#section--key-h-289-ref-null-props-children-init-classname-lang-meta-_owner-null-).

I also added the **Types** for this. I only included the basic **string** and **boolean** parameters since those are the only ones I needed at the moment.

Thanks again for your initial work and I hope my contribution can be helpful.

### CODE REVIEW

**Comment by pedro-lb** [comment]
Thanks for the PR @David-Melo!

One suggestion: maybe we could receive two arguments, `appId` and then `options` object (without `appId`), since all projects depending on this lib would break when updating.

I would prefer to avoid breaking changes, if possible.

What do you think?

**Comment by David-Melo** [comment]
You are right. I failed to take that into account. Keeping backwards compatibility is important. I will keep the first argument as the AppID and the second will be the options object. 

What do you think about making the second argument take either the options object OR a boolean for the "showNotififyButton". I could then check if its a boolean and convert it to the correct setting, but then also show a warning about deprecation.

"Warning, passing a boolean as a second parameter is deprecated." Or something like that. That way It will be fully backwards compatible if someone is passing in TRUE for that second parameter.

**Comment by pedro-lb** [comment]
I think the lib is quite new and `showNotifyButton` isn't a good solution overall. Your solution with a config object is much, much better.

The `appId` as the first argument is important, but we could do a new release informing users that `showNotifyButton` doesn't exist anymore and it should be passed in an object.

I'll review the code as soon as this is finished in this PR and we'll merge it and publish a new version.

**Comment by David-Melo** [comment]
Hello, I have implemented the changes and tested them locally. The first parameter works like before now and now the second parameter is the config object that will get parsed into the additional options.

**Comment by pedro-lb** [comment]
Hey there @David-Melo!

I noticed some problems:

- The code is breaking our current eslint rules, this should be fixed. Are you linting your repo?

- The method `mapOptionsObject` is too complex, violating base principles such as KISS.

I can take a look at this if you don't have the time, just tell me.

**Comment by David-Melo** [comment]
Hi, My apologies for the conflicts. I did not have linting setup when I made the changes. Do you have a specific lint file you use? 

I agree that the mapOptionsObject might be more complex than it needs to be, it was a bit over-engineered to accept more complex configs than what it would need at the moment.

**Comment by pedro-lb** [comment]
The package has eslint configured - you just need to have the eslint extension installed in your vscode (or other text editor).

https://github.com/pedro-lb/react-onesignal/blob/master/package/.eslintrc.js

File: package/src/oneSignal.ts [hunk 1]
```diff
@@ -10,20 +10,48 @@ const DEFAULT_MODULE_SCRIPT_ID = 'react-onesignal-module';
  */
 const ONE_SIGNAL_SCRIPT_SRC = 'https://cdn.onesignal.com/sdks/OneSignalSDK.js';
 
+/**
+ * Maps The Options Object Into A String For The Module Script
+ * @param o the options object
+ * @param indent 
+ */
+const mapOptionsObject = (o: any, indent: number = 0) => {
```
Comment (gax97) [inline package/src/oneSignal.ts:18]: Correct me if I am wrong, but wouldn't JSON.stringify(o, null, 1) also do the work?
If you don't want opening and closing braces you can do JSON.stringify(o, null, 1).slice(1).slice(0,-1).

Comment (pedro-lb) [reply package/src/oneSignal.ts:18]: Very good @gax97!

This would do the job and it's MUCH simpler :)

**Comment by pedro-lb** [comment]
Hey there @David-Melo,

I've merged your pull request and commited a fix to it. There were some issues:

- Numbers were being added as strings (e.g. `"42"`)
- Functions were being ignored, but the key was being added, invalidating the object (e.g. `myFunction: ,`)

You can check the changes at: 

https://github.com/pedro-lb/react-onesignal/commit/2d073151635e8d87e435f65e062516c447508f5d

**Comment by pedro-lb** [comment]
New release: https://github.com/pedro-lb/react-onesignal/releases/tag/1.1.16

**Pull request merged**